src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.CoreCLR.targets

```diff
@@ -12,4 +12,17 @@ This file contains the CoreCLR-specific MSBuild logic for .NET for Android.
     <_AndroidRuntimePackRuntime>CoreCLR</_AndroidRuntimePackRuntime>
   </PropertyGroup>
 
+  <Target Name="_CLRUseLocalRuntimePacks" AfterTargets="ResolveFrameworkReferences"
+          Condition=" '$(_CLRLocalRuntimePath)' != '' And '$(_AndroidRuntime)' == 'CoreCLR' ">
+    <PropertyGroup>
+      <_DotNetRuntimeRepo>$(_CLRLocalRuntimePath)</_DotNetRuntimeRepo>
+      <_DotNetRuntimeConfiguration Condition=" '$(_DotNetRuntimeConfiguration)' == '' ">Release</_DotNetRuntimeConfiguration>
+    </PropertyGroup>
+    <ItemGroup>
+      <!-- update runtime pack to local build -->
+      <ResolvedRuntimePack
+          PackageDirectory="$(_DotnetRuntimeRepo)/artifacts/bin/microsoft.netcore.app.runtime.$(RuntimeIdentifier)/$(_DotNetRuntimeConfiguration)"
+          Condition=" '%(ResolvedRuntimePack.FrameworkName)' == 'Microsoft.NETCore.App' And Exists('$(_DotnetRuntimeRepo)/artifacts/bin/microsoft.netcore.app.runtime.$(RuntimeIdentifier)/$(_DotNetRuntimeConfiguration)') " />
+    </ItemGroup>
+  </Target>
```

Comment:
We use this approach with Mono as well, if this is going to be accepted as part of the SDK, I suggest we make it runtime-agnostic